api/controller/assignments.js

```diff
@@ -0,0 +1,904 @@
+                        error: "An error has occurred",
+                        code: 10
+                    });
+                } else {
+                    if (result && result.result && result.result.n == 1) {
+                        res.status(200).send({
+                            id: req.params.assignmentId
+                        });
+                    } else {
+                        res.status(401).send({
+                            error: "Inexisting assignment id",
+                            code: 23
+                        });
+                    }
+                }
+            });
+    });
+};
+
+// update assignment
+exports.updateAssignment = function (req, res) {
+    //validate
+    if (!mongo.ObjectID.isValid(req.params.assignmentId)) {
+        return res.status(401).send({
+            'error': "Invalid assignment id",
+            'code': 35
+        });
+    }
+    if (!req.params.assignmentId) {
+        return res.status(400).send({
+            'error': "Assignment id is not defined",
+            'code': 22
+        });
+    }
+    var assignmentId = req.params.assignmentId;
+    var assignment = JSON.parse(req.body.assignment);
+    //add timestamp
+    assignment.timestamp = +new Date();
+    //find assignment by id
+    db.collection(assignmentCollection, function (err, collection) {
+        //count data
+        collection.countDocuments(
+            {
+                '_id': {
+                    $ne: new mongo.ObjectID(assignmentId)
+                },
+                'name': new RegExp("^" + assignment.name + "$", "i")
+            }
+            , function (err, count) {
+                if (err) {
+                    res.send({ 'error': 'An error has occurred in finding assignment' });
+                } else {
+                    if (count == 0) {
+                        collection.updateOne(
+                            {
+                                _id: new mongo.ObjectID(req.params.assignmentId)
+                            },
+                            {
+                                $set: assignment
+                            },
+                            {
+                                safe: true,
+                            },
+                            function (err, result) {
+                                if (err) {
+                                    return res.status(500).send({
+                                        error: "An error has occurred",
+                                        code: 10
+                                    });
+                                } else {
+                                    if (result && result.result && result.result.n == 1) {
+                                        res.send({
+                                            id: assignmentId
+                                        });
+                                    } else {
+                                        res.status(401).send({
```

Comment:
Sometimes you have used return with res.send, sometimes you haven't. Make it consistent.